massa-execution/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![feature(map_first_last)]
```

Comment:
What does that feature ?

Reply:
> I feel like I read the same thing multiple times but slightly different, Would be nice if at some point we take some time to clean up the code before merging it

Can you give more details ?

Reply:
> What does that feature ?

allows accessing the first and last elments of a btreemap/btreeset